### PR TITLE
[appmesh-controller] Use cert-manager v1 APIs, if available

### PR DESCRIPTION
**Pull request by mlazzaro-better**
### Issue

The `cert-manager.io/v1alpha2` APIs have been deprecated since v1.4 of cert manager, and have been fully removed as of v1.7 ([link to relevant docs](https://cert-manager.io/docs/installation/upgrading/remove-deprecated-apis/)).

Because of this, when attempting to use the chart on a cluster that's been upgraded to 1.7.x, the process fails with a error message like:

`Error: unable to build kubernetes objects from release manifest: [unable to recognize "": no matches for kind "Certificate" in version "cert-manager.io/v1alpha2", unable to recognize "": no matches for kind "Issuer" in version "cert-manager.io/v1alpha2"]`

### Description of changes

This PR updates the `appmesh-controller` chart to use the `cert-manager.io/v1` APIs if available, and then falls back to using the `v1alpha2` APIs if not, which should make the chart compatible with both older and newer cert-manager installations.

### Checklist
- [x] Added/modified documentation as required (such as the `README.md` for modified charts)
- [x] Incremented the chart `version` in `Chart.yaml` for the modified chart(s)
- [x] Manually tested. Describe what testing was done in the testing section below
- [x] Make sure the title of the PR is a good description that can go into the release notes

### Testing

Tested via tearing down an existing App Mesh installation on a cluster that had been upgraded to use `cert-manager` version 1.7.1 (and was previously failing, with the above error). After deleting, app mesh was re-deployed to the cluster successfully (using a local copy of this branch, rather than the remote repo).

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

**Comment by mlazzaro-better** [comment]
Hi, this is my first attempt at contributing to the repo, so please let me know if there's anything that I missed, or other into that needs to be added here!

**Comment by gitfool** [comment]
@achevuru This is blocking me due to helm chart dependency hell. Please review and merge. 🙏

**Comment by gitfool** [comment]
@bendu I don't suppose you can review (and approve) this too? 😁

**Comment by bendu** [comment]
@gitfool Will take a look. Again, i don't have write perms either...

**Review by bendu — APPROVED** [review]
@mlazzaro-better I've approved, but will need to find someone who as write access to merge.


seal of approval

**Comment by bendu** [comment]
Can someone help merge? This is blocking use of cert manager 1.7+. I don't have write permissions.
@kiranmeduri @achevuru @cgchinmay

**Comment by bendu** [comment]
@mlazzaro-better The PR has conflicts that need to be resolved, but change looks good.

**Review by jaypipes — APPROVED** [review]
:+1: seems reasonable to me, thank you @mlazzaro-better!

File: stable/appmesh-controller/Chart.yaml
Comment (jaypipes) [inline stable/appmesh-controller/Chart.yaml:4]: Looks like current version is 1.4.6 as of April 13: https://github.com/aws/eks-charts/blob/43abf55028ff65874fcd951ea7dc2f8170278c4d/stable/appmesh-controller/Chart.yaml#L4

You need to rebase and up this to 1.4.7 :)

**Review by jaypipes — CHANGES_REQUESTED** [review]

File: stable/appmesh-controller/Chart.yaml
Comment (gitfool) [reply stable/appmesh-controller/Chart.yaml:4]: @jaypipes I'm curious, since it's just the chart version, can't the maintainers do the rebase and patch the chart version?

Comment (jaypipes) [reply stable/appmesh-controller/Chart.yaml:4]: @gitfool yes, the maintainers could do the rebase and patch the chart version.

Comment (jaypipes) [reply stable/appmesh-controller/Chart.yaml:4]: Normally I would do this if GH would let me rebase in the UI... not in front of my desktop at the moment so can't use command line...

Comment (gitfool) [reply stable/appmesh-controller/Chart.yaml:4]: FWIW, it does allow rebase now; see [more ways to keep your pull request branch up-to-date](https://github.blog/changelog/2022-02-03-more-ways-to-keep-your-pull-request-branch-up-to-date).

Comment (bendu) [reply stable/appmesh-controller/Chart.yaml:4]: Let me see if I can take care of rebasing. Thanks @jaypipes for giving me write access

Comment (bendu) [reply stable/appmesh-controller/Chart.yaml:4]: I did the [rebase](https://github.com/bendu/eks-charts/commit/1b55ad0693d9cca5b176af16c6100a90c75a706e) Should I just push the change straight to master or make another PR?

Comment (gitfool) [reply stable/appmesh-controller/Chart.yaml:4]: @bendu do you have enough permission [allowing changes to a pull request branch created from a fork](https://docs.github.com/en/pull-requests/collaborating-with-pull-requests/working-with-forks/allowing-changes-to-a-pull-request-branch-created-from-a-fork)? In which case you would update the original branch and force push before approving and merging.

**Comment by bendu** [comment]
EKS charts is not the source of truth for the app mesh controller helm chart. I have created a PR  aws/aws-app-mesh-controller-for-k8s#577 to backfill this.

**Pull request closed without merging**
